CMakeLists.txt

```diff
@@ -1,5 +1,17 @@
 cmake_minimum_required(VERSION 3.29)
 
+function(append_platform_file sources platform path)
+    file(GLOB files "${path}/src/platform/*.c")
+    foreach(file ${files})
+        string(TOUPPER "${file}" file_upper)
+        if (${file_upper} MATCHES ${platform})
+            list(APPEND ${sources} ${file})
+            set(${sources} "${${sources}}" PARENT_SCOPE)
```

Comment:
This funny bit of CMake logic allows you to pass in a list,
append to that list,
then ensure that the list is updated outside the scope of the function
(thing pass by reference in C 💯 )

Reply:
So if I call `setup_bm_ip_stack(FLUGEL "${BM_CORE_FLUGEL_INCLUDES}")` does this add `bm_core/src/platform/foo_bar_flugel.c` and `bm_core/src/platform/baz_flugel_quux.c` but not `bm_core/src/platform/bm_lwip.c`? Am I understanding that correctly?

If so, we should explain in the docs and probably also in a comment here something for integrators like, "The platform key should be a case-insensitive substring within all the file names you want to add to sources."

Reply:
Correct.
I like that and I agree. I will add it in!